### PR TITLE
feat: use bit flags for user privilege check

### LINKED ISSUE #7097: feat: use bit flags for user privilege
### Is your feature request related to a problem? Please describe.

Currently in the meta and frontend user privileges are stored and used as `Vec`. We can use bit flag instead and optimize the relevant methods.

- Action definitions: [proto/user.proto](https://github.com/risingwavelabs/risingwave/blob/main/proto/user.proto)

- Related functions: 
    1. functions in frontend like `check_privilege`.
    2. functions in https://github.com/risingwavelabs/risingwave/blob/main/src/meta/src/manager/catalog/user.rs

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by yezizp2012**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Resolve #7097 .

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/common/Cargo.toml [hunk 1]
```diff
@@ -38,6 +38,7 @@ easy-ext = "1"
 educe = "0.4"
 either = "1"
 enum-as-inner = "0.6"
+enumflags2 = { version = "0.7.8" }
```
Comment (xxchan) [inline src/common/Cargo.toml:41]: Why not use the `bitflags` crate which we already used?

Comment (yezizp2012) [reply src/common/Cargo.toml:41]: Oh, I was planning to use `bitflags` as described in that issue, but found `enumflags2` somewhere else which is more readable, easier to transform between `PbAction` and derive display along with each enum.

Comment (xxchan) [reply src/common/Cargo.toml:41]: I slightly prefer no to introduce new dependencies, in order to reduce the risk of supply chain security, and also reduce compilation time.

Not a requirement. You decide it.

**Review by xxchan — APPROVED** [review]
cargo.lcok

File: src/common/Cargo.toml (hunk 1 above)
Comment (yezizp2012) [reply src/common/Cargo.toml:41]: Then we have to write many ugly code as follows if using `bitflags`:
```rust
impl Display for AclMode {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        if self == AclMode::Insert {
            write!(f, "a")
        }
        //....
    }
}

impl From<AclMode> for PbAction {
    fn from(mode: AclMode) -> Self {
        if mode == AclMode::Insert {
            return PbAction::Insert;
        }
        //....
    }
}
```
Let's introduce it this time. Not next time. 🥵

**Pull request merged**